anomalib/models/components/feature_extractors/torchfx.py

```diff
@@ -0,0 +1,155 @@
+                If the name is provided, the model is loaded from torchvision. Otherwise, the model class can be
+                provided and it will try to load the weights from the provided weights file.
+            return_nodes (Iterable[str]): List of layer names of the backbone to which the hooks are attached.
+                You can find the names of these nodes by using ``get_graph_node_names`` function.
+            weights (Optional[Union[WeightsEnum,str]]): Weights enum to use for the model. Torchvision models require
+                ``WeightsEnum``. These enums are defined in ``torchvision.models.<model>``. You can pass the weights
+                path for custom models.
+            requires_grad (bool): Models like ``stfpm`` use the feature extractor for training. In such cases we should
+                set ``requires_grad`` to ``True``. Default is ``False``.
+
+        Returns:
+            Feature Extractor based on TorchFX.
+        """
+        backbone_model = self.get_backbone_class(backbone.class_path)
+        if isinstance(weights, WeightsEnum):  # torchvision models
+            feature_extractor = create_feature_extractor(
+                backbone_model(weights=weights, **backbone.init_args), return_nodes
+            )
+        else:
+            backbone_model = backbone_model(**backbone.init_args)
+            if weights is not None:
+                assert isinstance(weights, str), "Weights should point to a path"
+                backbone_model.load_state_dict(torch.load(weights)["state_dict"])
+            feature_extractor = create_feature_extractor(backbone_model, return_nodes)
+
+        if not requires_grad:
+            feature_extractor.eval()
+            for param in feature_extractor.parameters():
+                param.requires_grad_(False)
+
+        return feature_extractor
+
+    def get_backbone_class(self, backbone: str) -> nn.Module:
+        """Get the backbone class from the provided path.
+
+        If only the moodel name is provided, it will try to load the model from torchvision.
+
+        Args:
+            backbone (str): Path to the backbone class.
+
+        Returns:
+            Backbone class.
+        """
+        try:
+            if len(backbone.split(".")) > 1:
+                # assumes that the entire class path is provided
+                models = importlib.import_module(".".join(backbone.split(".")[:-1]))
+                backbone_model = getattr(models, backbone.split(".")[-1])
+            else:
+                models = importlib.import_module("torchvision.models")
+                backbone_model = getattr(models, backbone)
+        except ModuleNotFoundError as exception:
+            raise ModuleNotFoundError(
+                f"Backbone {backbone} not found in torchvision.models and not found in {backbone} module either"
+            ) from exception
+
+        return backbone_model
+
+    def __call__(self, inputs: Tensor) -> Dict[str, Tensor]:
```

Comment:
It would be an idea to decide on the name of the variable. We use `input`, `inputs` and `input_tensor` interchangeably. As far as I know, PyTorch prefers `input`. It might be an idea to settle down on a single variable name

Reply:
How about `inputs` as we type it as `Tensor`. `input` might clash with built in `input` method.